### PR TITLE
fix(modal): fix spacing and overriding styles in the docsite

### PR DESCRIPTION
**Pull request by josedialpad**
## Description
Fixes the overriding styles in the docsite.
It also fixes the spacing in the variant when no footer is provided.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: lib/build/less/components/modal.less [hunk 1]
```diff
@@ -139,18 +137,18 @@
 //  ----------------------------------------------------------------------------
 .d-modal__header {
   margin: 0 !important;
-  padding: var(--su24) var(--su24) 0;
+  padding: var(--su24) var(--su24) 0 !important;
   color: var(--modal-header--fc);
   font-weight: var(--fw-bold);
-  font-size: var(--fs24);
+  font-size: var(--fs24) !important;
```
Comment (francisrupert) [inline lib/build/less/components/modal.less]: Is this a bit too brute-force? I thought the real culprit was VuePress' over-specific CSS selectors leaking into our components? 

Can VuePress' CSS be updated rather than updating the DT component itself?

The age-old "avoid `!important`" is the go-to thinking here, particularly since we're deliberate about `!important` for Utilities.

Comment (josedialpad) [reply lib/build/less/components/modal.less]: Yeah I agree and the first approach I've tried was to adapt the styles not to override the modal's style, and I found that we're overriding the styles here https://github.com/dialpad/dialtone/blob/staging/docs/.vuepress/styles/palette.scss#L11-L24 that's in our side and can be modified, but also the Vuepress default theme (which we're using) has a set of styles we cannot modify since they're coming from the `@vuepress` package https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/theme-default/src/client/styles/layout.scss#L123-L152

What I found in Vuepress v1 is the `eject` command to copy the default theme source into the project so that way we can customize it but that implies the theme will stop to receive any update and will require manual update. Also the `eject` command was removed in the v2 https://v2.vuepress.vuejs.org/guide/migration.html#cli-change:~:text=%23-,eject%20command,-Removed.

Comment (braddialpad) [reply lib/build/less/components/modal.less]: Is there a way we can change the specificity of the CSS in this component to make this work without important? 

Additionally, looks like `.theme-default-content` is the problematic class, is there a way we can exclude `.theme-default-content` from affecting our examples at the Vuepress level? Is it possible to use `:not()` or something like that?

Comment (josedialpad) [reply lib/build/less/components/modal.less]: > Additionally, looks like .theme-default-content is the problematic class, is there a way we can exclude .theme-default-content from affecting our examples at the Vuepress level? Is it possible to use :not() or something like that?

I couldn't find an easy way, to do it it requires changing the layout or extending the default theme.

> Is there a way we can change the specificity of the CSS in this component to make this work without important?

Updated to fix with specificity https://github.com/dialpad/dialtone/pull/635/commits/9aed7077a8e14cd4e973b7a288b32e4211423dc1

Basically, we need to override the `.d-modal__header` class just giving higher specificity to win the battle against the styles coming from the default theme that's not possible to modify https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/theme-default/src/client/styles/layout.scss#L123-L152

Comment (francisrupert) [reply lib/build/less/components/modal.less]: Quite a workaround, nicely done. While it's a blatant code smell...

image

...it's as reasonable a workaround as any, better than `!important`. Hopefully we won't have to do this too often going forward, let alone if we haven't caught other instances of VuePress styles leaking in.

Very unfortunate VuePress didn't realize how much their style approach _will_ leak in. A significant redesign/rearchitecture of doc site is on the backburner, but nowhere near a priority. 

I approve this change, what say you @braddialpad?

Comment (braddialpad) [reply lib/build/less/components/modal.less]: Yep I'm okay with this.

**Review by braddialpad — APPROVED** [review]
